Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 1427e89fce38b3272d2a2ab72c96fc773a153004
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/PR-13213/HPCC-Platform/esp/bindings/SOAP/Platform/soapparam.cpp:71:9: error: \xe2\x80\x98reteurn\xe2\x80\x99 was not declared in this scope
make[2]: *** [esp/protocols/http/CMakeFiles/esphttp.dir/build.make:245: esp/protocols/http/CMakeFiles/esphttp.dir/__/__/bindings/SOAP/Platform/soapparam.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:10368: esp/protocols/http/CMakeFiles/esphttp.dir/all] Error 2
make: *** [Makefile:163: all] Error 2

[Build error:](https://gist.github.com/f92bf0f72c25be8d351edbffe69ab8e4)
Build log [Build log](https://gist.github.com/HPCCSmoketest/f92bf0f72c25be8d351edbffe69ab8e4#file-result-19-12-09-14-56-17-log)

--- a/esp/bindings/SOAP/Platform/soapparam.cpp
+++ b/esp/bindings/SOAP/Platform/soapparam.cpp
@@ -68,8 +68,9 @@ bool BaseEspParam::unmarshall(IRpcMessage &rpc_call, const char *tagname, const 
         if (optGroup && rpc_call.queryContext())
             rpc_call.queryContext()->addOptGroup(optGroup);
         isNil = false;
+        return true;
     }
-    return isNil;
+    return false;
 }
 
 bool BaseEspParam::unmarshall(IEspContext* ctx, CSoapValue &soapval, const char *tagname, const char* optGroup)
@@ -79,8 +80,9 @@ bool BaseEspParam::unmarshall(IEspContext* ctx, CSoapValue &soapval, const char 
         if (ctx && optGroup)
             ctx->addOptGroup(optGroup);
         isNil = false;
+        return true;
     }
-    return isNil;
+    return false;
 }
 
 bool BaseEspParam::unmarshall(IEspContext* ctx, IProperties &params, MapStrToBuf *attachments, const char *tagname, const char *basepath, const char* optGroup, const char *xsdtype, const char *prefix)
@@ -92,11 +94,12 @@ bool BaseEspParam::unmarshall(IEspContext* ctx, IProperties &params, MapStrToBuf
 
     if (updateValue(params.queryProp(path.str())))
     {
-        isNil = false;
         if (ctx && optGroup)
             ctx->addOptGroup(optGroup);
+        isNil = false;
+        return true;
     }
-    return isNil;
+    return false;
 }
 
 void SoapStringParam::toXMLValue(StringBuffer &s, bool encode)

